### DISCUSSION TITLE
Unable to pip install PyMuPDF (running on new M1pro Macbook)

### DISCUSSION THREAD

**Discussion by dutchwine85** [pymupdf/PyMuPDF#1486] [Installation Topics]
I am trying to install PyMuPDF on a new M1Pro Macbook but I get a large error message each time. My input is -
`pip3 install PyMuPDF`

I have tried this several times but the output is the same.
The error is very long but some key pieces appear to be -

`fitz/fitz_wrap.c:1756:7: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]`

`/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
      Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
      ^`

`fitz/fitz_wrap.c:2755:10: fatal error: 'fitz.h' file not found
  #include <fitz.h>
           ^~~~~~~~
  2 warnings and 1 error generated.
  error: command 'clang' failed with exit status 1`

I have already installed fitz with no issues. Are there any other packages I need to install prior to retrying PyMuPDF?
Thanks

**Comment by JorjMcKie** [comment]
This is a duplicate issue, but no bug:
There are no wheels for M1 platforms (and there will not be any time soon). So you must install from sources - which means you must install MuPDF first, then PyMuPDF.
I will convert this to a Discussions item and put it in the respective category.

**Comment by JorjMcKie** [comment]
Please see [here](https://github.com/pymupdf/PyMuPDF/discussions/875).

**Comment by jr2173** [comment]
Hi @JorjMcKie - first want to thank you for all your efforts with this amazing library.

I noticed that `aarch64` wheels were bundled with [1.18.15](https://pypi.org/project/PyMuPDF/1.18.15/#files) and do allow PyMuPDF to be installed effortlessly on this architecture.  After reading through most of the comments here relating to this I can't pinpoint why those wheels stopped being included.  Can you perhaps shed some light on why and what might be needed to include them in the CI build?

Thanks

**Comment by JorjMcKie** [comment]
@jr2173 - I thought the `aarch64` only work on Linux ARM?
Well I can restart generating them. There are the following issues with them however:
* There currently only is the Github Actions which support their creation at all, and this happens under emulation
* I must suppress building Tesseract OCR support with them, because otherwise the 90 minute duration limit is exceeded

So since v1.19.0 I have put these generations under hiat and wait until someone complains ... 😒

**Comment by jr2173** [comment]
Sorry I jumped the gun a bit, I'm actually using Docker where the `aarch64` wheel from 1.18.15 works fine within a Linux-based container on an M1 Mac.  On the Mac side, which is not as critical to us, it does want to compile from source and requires mupdf swig freetype deps.

Still exploring how to streamline this as we're seeing more M1 Macs being used in our dev cycle, perhaps a separate Dockerfile for `aarch64` that builds from source for newer releases.  Though if you could include those wheels, with the caveats, it'd be incredibly helpful!

**Comment by JorjMcKie** [comment]
I can make a set of wheels based on the 1.19.5 pre-version more or less now. Give me a day or so.
I will let you know when you can dwonload them from the [wheels repo](https://github.com/pymupdf/PyMuPDF-wheels/tree/master/wheelhouse).

**Comment by felixgao** [comment]
I am facing the same issue and the current version is 1.19.6, so I guess there still isn't a M1 based wheel?

**Comment by xy2i** [comment]
Same here, 2024 on mac M1

**Comment by julian-smith-artifex-com** [comment]
We have wheels for MacOS/arm64 these days.

So please create a new issue for the problem you're seeing, using https://github.com/pymupdf/PyMuPDF/issues/new?template=bug_report.yml.

[I'll close this discussion as it's very old and out of date now.]
